scripts/tools/EXT_Exp_MetricProgrammable.rst

```diff
@@ -0,0 +1,123 @@
+
+Sample Code
+------------
+
+The following pseudo-code demonstrates how programmable metrics could be enumerated and custom values could be set to the metric parameters.
+
+.. parsed-literal::
+
+    ${t}_metric_handle_t * metricHandles = null_ptr;
+    uint32_t metricHandleCount = 0;
+    ${t}_metric_group_handle_t metricGroup;
+
+    // Query and Get metric programmable handles
+    uint32_t programmableCount = 0;
+    ${t}MetricProgrammableGetExp(device, &programmableCount, nullptr);
+    ${t}_metric_programmable_exp_handle_t * metricProgrammableHandles = allocate(sizeof(${t}_metric_programmable_exp_handle_t) * programmableCount);
+    ${t}MetricProgrammableGetExp(device, &programmableCount, metricProgrammableHandles);
+
+    // Create metrics from metric programmable handles
+    for(uint32_t i = 0; i < programmableCount; i++){
+
+        ${t}_metric_programmable_exp_handle_t programmableHandle = metricProgrammableHandles[i];
+        ${t}_metric_programmable_exp_properties_t programmableProperties{};
+        ${t}MetricProgrammableGetPropertiesExp(programmableHandle, &programmableProperties);
+
+        // Choose programmable handle of interest
+        if(strcmp(programmableProperties.name, "EU_ACTIVE" ) == 0){
+
+            // Get Parameter info
+            ${t}_metric_programmable_param_info_exp_t * paramInfo = allocate (sizeof(${t}_metric_programmable_param_info_exp_t) * programmableProperties.parameterCount);
+            ${t}MetricProgrammableGetParamInfoExp(programmableHandle, programmableProperties.parameterCount, paramInfo);
+
+            // Get Parameter Value info for the 0th parameter
+            ${t}_metric_programmable_param_value_info_exp_t * paramValueInfo = allocate(sizeof(${t}_metric_programmable_param_value_info_exp_t) * paramInfo[0].paramValueInforiptionCount);
+            ${t}MetricProgrammableGetParamValueInfoExp(programmableHandle, 0, paramInfo[0].valueInfoCount, paramValueInfo);
+
+            // Setting value for the 0th parameter
+            ${t}_metric_programmable_param_value_exp_t parameterValue;
+            parameterValue.value = paramValueInfo[0].valueInfo.ui64;
+
+            // Create Metric
+            ${t}_metric_handle_t metricHandle;
+            char metricName[ZET_MAX_METRIC_NAME] = "eu_active_minimum";
+            char metricDescription[ZET_MAX_METRIC_DESCRIPTION] = "eu_active_minimum_desc";
+            ${t}MetricCreateFromProgrammableExp(programmableHandle, &parameterValue, 1, metricName, metricDescription, &metricHandleCount, null_ptr);
+            metricHandles = allocate(sizeof(${t}_metric_handle_t) * metricHandleCount);
+            ${t}MetricCreateFromProgrammableExp(programmableHandle, &parameterValue, 1, metricName, metricDescription, &metricHandleCount, metricHandles);
+        }
+
+        //Create Metric Group from metrics
```

Comment:
seems this should be inside the `if(strcmp(programmableProperties.name, "EU_ACTIVE" ) == 0){`

Reply:
Yes .. the closing braces seem incorrect.
Will fix.
Thanks